File: app/channels/comment_channel.rb
class CommentChannel < ApplicationCable::Channel
  def subscribed
    @podcast = Podcast.find(params[:podcast_id])
    stream_for @podcast
  end

  def receive(data)
    user = User.find(data['user_id'])
    comment = user.comments.create!(text: data['text'], podcast_id: data['podcast_id'])
    
    CommentChannel.broadcast_to(
      comment.podcast,
      comment: comment.as_json,
      user: user.as_json.merge(icon_url: user.icon_url),
      created_at: comment.created_at.strftime('%Y-%m-%d %H:%M:%S')
    )
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/comment_channel.js
import consumer from "./consumer"

if(location.pathname.match(/\/podcasts\/\d/)){

  consumer.subscriptions.create({
    channel: "CommentChannel",
    podcast_id: location.pathname.match(/\d+/)[0]
  }, {

    connected() {
    },

    disconnected() {
    },

    received(data) {
      const html = `
        <div class="slack-message">
          <div class="slack-message-avatar">
            <img src="${data.icon_url}" alt="ユーザーアイコン">
          </div>
          <div class="slack-message-content">
            <div class="slack-message-header">
              <span class="slack-message-name">${data.user.nickname}</span>
              <span class="slack-message-time">${data.created_at}</span>
            </div>
            <div class="slack-message-text">${data.comment.text}</div>
          </div>
        </div>`;
      const comments = document.getElementById("comments")
      comments.insertAdjacentHTML('beforeend', html)
      const commentForm = document.getElementById("comment-form")
      commentForm.reset();

      comments.scrollTop = comments.scrollHeight;
    }
  })
}